src/schema/v2/me/user_flags.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  GraphQLError,
+  GraphQLFieldConfig,
+  GraphQLScalarType,
+  Kind,
+  ObjectValueNode,
+  print,
+  ValueNode,
+} from "graphql"
+import Maybe from "graphql/tsutils/Maybe"
+import { camelCase } from "lodash"
+import { ResolverContext } from "types/graphql"
+
+export const UserFlagsType = new GraphQLScalarType({
+  name: "UserFlagsType",
+  description:
+    "The user's user_flags. Values of keys in user_flags can be anything and keys may not be known beforehand.",
+  serialize: (value) => value,
+  parseValue: (value) => value,
+  parseLiteral: (ast, variables) => {
+    if (ast.kind !== Kind.OBJECT) {
+      throw new TypeError(
+        `Object cannot represent non-object value: ${print(ast)}`
+      )
+    }
+
+    return parseObject("Object", ast, variables)
+  },
+})
+
+const deriveAdditionalFlags = async (root, options, loaders) => {
+  const { collectorProfileLoader } = loaders
+  const computedFlags: { [key: string]: any } = {}
+
+  const result = await collectorProfileLoader(options)
+  const { bio, icon, other_relevant_positions } = result
+  const profileIsComplete =
+    root.profession && bio && icon && other_relevant_positions
+
+  if (!profileIsComplete) {
+    computedFlags.collector_profile_incomplete_at = new Date().toISOString()
```

Comment:
Don't we need to set `collector_profile_incomplete_at` to `null` if the user has seen the message? Even when setting the value to `null` the query will always return the timestamp.